### DISCUSSION TITLE
Is it possible to get Event buffer on Virtualkeypad Web without LCD keypad?

### DISCUSSION THREAD

**Discussion by bligeti** [taligentx/dscKeybusInterface#205] [dscKeybusInterface Discussion] [answered]
Hi,

I have a PC585 panel with 2 PC1555RKZ (LED) keypad. I'm using this library and the web Virtualkeypad for a while and with the 2.0 I was hoping I will get access to the event buffer which is not available due to the lack of LCD keypad.
Unfortunately the event buffer does not seem to work on Virtualkeypad in my case. Neither using the enter button nor going into the *6 menu and navigating with the arrows.
Any chance to get the event buffer without a LCD keypad?
Is there any setting of the panel which might be missing and results in this behaviour?

Thank you for the help in advance!

**Comment by kricon** [comment, marked as answer]
> Unfortunately the event buffer does not seem to work on Virtualkeypad in my case. Neither using the enter button nor going into the *6 menu and navigating with the arrows.
> Any chance to get the event buffer without a LCD keypad?
> Is there any setting of the panel which might be missing and results in this behaviour?

You're supposed to press `*` followed by `6` then master code, and then press "`Enter`". It should bring you in event buffer display.
Navigating with arrows works only while viewing event buffer, it **doesn't** work anywhere else, that's normal behaviour.
Viewing event buffer should work even without LCD keypad, but only on partition 1. Maybe pauseZones isn't called because too-quickly pressing digits or Web server latency (from my experience, `VirtualKeypad-Web` example was VERY laggy and I had problems with zone lights following program/memory lights because of it and probably pauseZones didn't get called because of lag or bad wifi connection as it was quite far from the access point).
There is no setting in the panel programing which can prohibit entering or viewing event buffer, as long as you know master code and can enter *6 menu.

**Reply by bligeti** [reply]
Thank you for your answer @kricon! I think I also tried this way without luck, but confirming that it should be possible and also the correct method helped me to do some troubleshooting on the code. Strangly the pausedZones flag was not working for me until I flag it to keep it in DRAM. So it is working now. Thanks!
